src/unix/solarish/mod.rs

```diff
@@ -2154,6 +2201,9 @@ pub const PTHREAD_RWLOCK_INITIALIZER: pthread_rwlock_t = pthread_rwlock_t {
 pub const PTHREAD_MUTEX_NORMAL: ::c_int = 0;
 pub const PTHREAD_MUTEX_ERRORCHECK: ::c_int = 2;
 pub const PTHREAD_MUTEX_RECURSIVE: ::c_int = 4;
+#[cfg(target_os = "illumos")]
+pub const PTHREAD_MUTEX_DEFAULT: ::c_int = 8;
```

Comment:
Until we (illumos) make a decision about what the minimum support illumos version is, this change to PTHREAD_MUTEX_DEFAULT cannot be made